src/Bicep.Core/Semantics/Binder.cs

```diff
@@ -48,6 +48,41 @@ public Binder(SyntaxTree syntaxTree, ISymbolContext symbolContext)
         public SyntaxBase? GetParent(SyntaxBase syntax)
             => syntaxTree.Hierarchy.GetParent(syntax);
 
+        public ImmutableArray<TSyntax> GetAllAncestors<TSyntax>(SyntaxBase syntax) where TSyntax : SyntaxBase
```

Comment:
>GetAllAncestors [](start = 39, length = 15)

Can we put these on `SyntaxHierarchy` or make them extension methods on it? The logic is related to that. Or could introduce an interface that includes just the GetParent method and then the extensions methods could work on either the `SyntaxHierarchy` class or the binder.

Reply:
It's ultimately better for implementation-heavy things like this to live on a class like `SyntaxHierarchy` unless you really want it to be a replaceable abstraction (extension method approach). `SyntaxHierarchy` doesn't optimize these oprerations today, but you could image it doing so in the future if its valuable.

Reply:
I put something together here with default interface methods. LMK if you still feel strongly about having these be extensions. DIMs are a little clunky, but they allow you to specialize the implementation later without a binary compat break.